x/posts/internal/keeper/handler.go

```diff
@@ -132,17 +134,17 @@ func handleMsgAddPostReaction(ctx sdk.Context, keeper Keeper, msg types.MsgAddPo
 	}
 
 	// Create and store the reaction
-	reaction := types.NewReaction(msg.Value, msg.User)
-	if err := keeper.SaveReaction(ctx, post.PostID, reaction); err != nil {
+	reaction := types.NewPostReaction(msg.Value, msg.User)
```

Comment:
I think there's a missing check to make sure that the `msg.Value` is a registered reaction before saving it

Reply:
this check is performed inside the `SavePostReaction` keeper method to perform the check even on genesis data